tldw_Server_API/app/core/DB_Management/Collections_DB.py

```
@@ -1038,6 +1039,38 @@ def mark_highlights_stale_if_content_changed(self, item_id: int, new_content_has
+            candidate = candidate.expanduser()
+            if candidate.is_absolute():
+                resolved = candidate.resolve(strict=False)
+            else:
+                resolved = (base_resolved / candidate).resolve(strict=False)
+        except Exception as exc:
+            logger.warning(f"outputs: invalid output path {path_value}: {exc}")
+            raise ValueError("invalid_path") from exc
+        try:
+            if not resolved.is_relative_to(base_resolved):
+                logger.warning(f"outputs: output path outside base dir: {resolved}")
+                raise ValueError("invalid_path")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)